scripts/set_bary_helio_times.py

```diff
@@ -111,25 +109,17 @@ def set_bary_helio_times(filename, jwstpos=None):
                 times=mjdtimes
             )
         except Exception as exception:
-            logging.warning(
-                'Error in calculating times. Filling with invalid dates.'
-                '\nError is "{}"'.format(exception)
-            )
-            nulls = -1.0 * np.ones(len(tendtimeslist), dtype=np.float)
-            bcol = fits.Column(
-                name='bary_end_time', format='D', unit='MJD', array=nulls
-            )
-            hcol = fits.Column(
-                name='helio_end_time', format='D', unit='MJD', array=nulls
-            )
+            # Cleanup before exiting
+            hdul.flush()
+            hdul.close()
+            raise Exception('Error in calculating times.') from exception
```

Comment:
Better here would be something like:
```
        except Exception as exception:
            # Cleanup before exiting
            hdul.flush()
            hdul.close()
            raise exception
```
That way you are raising the exception that actually caused the failure, not a new, no-name exception.

Reply:
True, that I'm raising another exception, but the use of "from exception" does also cause the original exception (with its error message) to be "emitted", followed by the new one. So that way you get both the original error (whatever it was, like db access error, for example), plus a statement from set_bary_helio_times itself giving you a line number in the traceback. (Hey, I learned something about python 3 exceptions today!)

Reply:
Ah yes, getting the line in `set_bary_helio_times` is useful in the traceback.  🥇